wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/ModalDialog.js

```diff
@@ -235,7 +235,7 @@
 			}
 			if (data.options.closeOnEscape) {
 				overlay.on("keydown", function (event) {
-					if (event.which == 27) {
```

Comment:
By the way `which` is deprecated https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which, `key` is recommended instead https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key